### ISSUE TITLE
Default comparator for raw Collections

### ISSUE THREAD

**Issue by bartoszwalacik** [javers/javers#746]
Currently, JaVers skips comparing fields of Collection type

For example, in this class, `clients` field is ignored:

```java
public class Company {
    private String id; 
    private Collection<Person> clients;
    private List<Person> partners;
}
```
and only this warnign is logged. Expected behaviour would be using List comparator.  

```
10:19:40.332 [main] WARN  o.j.c.d.a.CollectionChangeFakeAppender - Collections: Field Collection<Person> clients; //declared in Company
are not equals but can't be compared. Raw Collection properties are not supported. Expected Set, List or any of their subclasses. JaVers uses different algorithms for comparing Sets and Lists and needs to know (statically) which one to test.
```

**Comment by bartoszwalacik** [comment]
released in 4.0.0-RC3

**Comment by baumgartner** [comment]
```
    @Test
    public void equalEntityClientTest() {
        Person e1 = new Person("james", "ignore this");
        List<Person> l1 = new ArrayList<Person>();
        l1.add(e1);
        Company le1 = new Company("1", null, l1, null);

        Person e2 = new Person("james", "");
        List<Person> l2 = new ArrayList<Person>();
        l2.add(e2);
        Company le2 = new Company("1", null, l2, null);
        Diff diff = javers.compare(le1, le2);

        System.out.println(diff);
        assertEquals(0, diff.getChanges().size());
    }
```
Works with ListCompareAlgorithm.Simple 
Assert fails with ListCompareAlgorithm.AS_SET
Exception with ListCompareAlgorithm.LEVENSHTEIN_DISTANCE 

```
java.lang.ClassCastException: java.util.Collections$UnmodifiableSet cannot be cast to java.util.List
	at org.javers.core.diff.appenders.levenshtein.LevenshteinListChangeAppender.calculateChanges(LevenshteinListChangeAppender.java:39)
	at org.javers.core.diff.appenders.levenshtein.LevenshteinListChangeAppender.calculateChanges(LevenshteinListChangeAppender.java:21)
	at org.javers.core.diff.appenders.CollectionAsListChangeAppender.calculateChanges(CollectionAsListChangeAppender.java:32)
	at org.javers.core.diff.appenders.CollectionAsListChangeAppender.calculateChanges(CollectionAsListChangeAppender.java:14)
	at org.javers.core.diff.appenders.CorePropertyChangeAppender.calculateChanges(CorePropertyChangeAppender.java:45)
	at org.javers.core.diff.DiffFactory.appendChanges(DiffFactory.java:151)
	at org.javers.core.diff.DiffFactory.appendPropertyChanges(DiffFactory.java:141)
	at org.javers.core.diff.DiffFactory.createAndAppendChanges(DiffFactory.java:123)
	at org.javers.core.diff.DiffFactory.create(DiffFactory.java:69)
	at org.javers.core.diff.DiffFactory.compare(DiffFactory.java:54)
	at org.javers.core.JaversCore.compare(JaversCore.java:173)
	at javers.test.JaversListTest.equalEntityClientTest(JaversListTest.java:37)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
```
Test-Project: https://github.com/baumgartner/javerstest/blob/master/src/test/java/javers/test/CollectionCompareTest.java

**Comment by bartoszwalacik** [comment]
Fixed in 5.0.1 for all list comparing algorithms.
